docs/rst/tutorials/micro/user_api/user_api_middleware.rst

```diff
@@ -0,0 +1,279 @@
+.. _tutorials_micro_user_middleware:
+
+.. TODO: In other tutorials:
+   - Custom allocators API
+   - Set custom transport API
+   - Error handling
+   - Discovery ?
+
+Middleware API
+========================
+
+.. contents:: Table of Contents
+    :depth: 2
+    :local:
+    :backlinks: none
+
+micro-ROS default rmw layer `rmw_microxrcedds_c <https://github.com/micro-ROS/rmw_microxrcedds>`_ offers a set of utilities to directly configure or interact with the underlying middleware.
+The exposed API detailed on this documentation can be used with the ``rmw_microxrcedds_c`` public header: ``#include <rmw_microros/rmw_microros.h>``.
+
+Transport configuration
+-----------------------
+
+Middleware provided transports can be configured at run-time, allowing the user to set the Agent address or the serial port identifier:
+
+.. code-block:: c
+
+    // Init RCL options and context
+    rcl_init_options_t init_options = rcl_get_zero_initialized_init_options();
+    rcl_context_t context = rcl_get_zero_initialized_context();
+    rcl_init_options_init(&init_options, rcl_get_default_allocator());
+
+    // Take RMW options from RCL options
+    rmw_init_options_t* rmw_options = rcl_init_options_get_rmw_init_options(&init_options);
+
+    // UDP transport configuration
+    char* agent_ip = "127.0.0.1";
+    char* agent_port = "8888";
+    rmw_uros_options_set_udp_address(agent_ip, agent_port, rmw_options);
+
+    // Serial transport configuration
+    char* serial_device = "/dev/ttyAMA0";
+    rmw_uros_options_set_serial_device(serial_device, rmw_options)
+
+.. note::
+
+   Transport must be enabled on build time using ``RMW_UXRCE_TRANSPORT_UDP`` or ``RMW_UXRCE_TRANSPORT_SERIAL`` options.
+
+Client Key configuration
+------------------------
+
+The client key will identify a micro-ROS client on the Agent side, implying that all the connected clients must have a different key.
+This value is set randomly on micro-ROS initialization, but it is also possible to set it manually:
+
+.. code-block:: c
+
+    // Init RCL options and context
+    rcl_init_options_t init_options = rcl_get_zero_initialized_init_options();
+    rcl_context_t context = rcl_get_zero_initialized_context();
+    rcl_init_options_init(&init_options, rcl_get_default_allocator());
+
+    // Take RMW options from RCL options
+    rmw_init_options_t* rmw_options = rcl_init_options_get_rmw_init_options(&init_options);
+
+    // Set RMW client key
+    uint32_t client_key = 0xBA5EBA11;
+    rmw_uros_options_set_client_key(0xBA5EBA11, rmw_options);
+
+This feature can be useful for reusing DDS entities already created on the agent side, as explained on this `Micro XRCE-DDS Deployment example <https://micro-xrce-dds.docs.eprosima.com/en/latest/getting_started.html#deployment-example>`_.
```

Comment:
Add a todo for adding a new tutorial on this and also add it to the list of future tutorials please

Reply:
https://github.com/eProsima/vulcanexus/pull/78/commits/346a5be6837282f7b23d35cfe1375a8f7aa19ecd

Also added to tutorials list